client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -63,6 +64,16 @@ export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
     return true;
   };
 
+  const parseError = (error: string): LocaleKey => {
+    switch (true) {
+      case error.indexOf('AdjustmentReasonNotProvided') !== -1:
```

Comment:
Sorry didn't get a change to review the reasons PR.

I am pretty strong against matching errors like this, I think it should be done via error variant.

An example of this:

https://github.com/openmsupply/open-msupply/blob/577f84f019e0d709613f9fbced59a12070e5389b/client/packages/host/src/components/Initialise/hooks.ts#L106-L118

https://github.com/openmsupply/open-msupply/blob/577f84f019e0d709613f9fbced59a12070e5389b/server/graphql/general/src/mutations/initialise_site.rs#L11-L14

I think both of these errors should be changed to be Structured Errors, similar to the above example but different Structured Errors (in one API returns stock line, so we can show extra info to the user), similar to:

Reply:
Both, as in AdjustmentReasonNotProvided as well?